### ISSUE TITLE
Move 'pkg/kubelet/cm/cpuset' to kubernetes/utils

### ISSUE THREAD

**Issue by iancoolidge** [kubernetes/kubernetes#112899]
### What would you like to be added?

I would like to use 'pkg/kubelet/cm/cpuset' as a library in a different project.
Is it possible to relocate it to kubernetes/utils?

### Why is this needed?

'cpuset' package is not staged as part of this package interface, it's a subpackage.

**Comment by BenTheElder** [comment]
Hi @cyclinder, has SIG node discussed if this is something they want to support yet?

**Comment by iancoolidge** [comment]
@BenTheElder When does that happen? Do I need to do anything to trigger that discussion?

**Comment by BenTheElder** [comment]
Typically SIG node sweeps the issues periodically but to propose feature work rather than reporting a bug I would bring it to their slack, mailinglist, or biweekly meeting for that particular sig.

As a general rule features for Kubernetes are tracked via KEPs at this point and the issue tracker is a bit overwhelming. Additionally bringing a particular issue to attention by other means cuts through the noise a bit.

**Comment by lavalamp** [comment]
I don't think you need SIG Node's permission to *copy* the package to utils, you need SIG Node to be on board to remove the existing one and change references.

**Comment by BenTheElder** [comment]
Well ... From an unrelated discussion today: utils has or should have a higher bar for maintaining compatibility of packages, vs unexported packages here.

I'd personally suggest input from Node regarding if this is a maintainable thing to export for external usage.

**Comment by lavalamp** [comment]
Sure, I agree with that.

**Comment by SergeyKanzhelev** [comment]
`MustParse` is never used in code. Perhaps it can be deleted instead of making it `panic`. 

Also functionality looks linux specific, perhaps when making a library it also needs to be explicit. Maybe not build it for Windows.

> I would like to use 'pkg/kubelet/cm/cpuset' as a library in a different project.

@iancoolidge I wonder what functionality you need the most out of it? Parsing? Asking to make sure there is no dependency on things like checkpointing format. I don't think it is something we need to make an API from.

**Comment by iancoolidge** [comment]
@SergeyKanzhelev I do not need 'MustParse', I am mostly just using it to parse cpuset ranges (read out of sysfs), manipulate them, and then emit stringified cpuset ranges for feeding to kernel command line arguments for nohz_full / isolcpus etc.

I don't think anything is OS specific in there, it doesn't actually do anything with sysfs. I don't know if cpusets are a useful construct on windows, but I don't see any reason to exclude the library there.

**Comment by liggitt** [comment]
yeah, if it's just string manipulation / construction, that seems useful to be able to do regardless of the OS the binary is running on (imagine a CLI tool running on an arbitrary OS that is used to construct configurations to ship off to a linux machine)

**Comment by iancoolidge** [comment]
So, any oppoition to this? I'm happy to start work on this, or if someone else is interested in doing it now-ish, happy for help.

**Comment by klueska** [comment]
I am happy to take a look at this from a sig-node perspective and have assigned myself to https://github.com/kubernetes/utils/pull/267

I will begin looking at it more seriously once this is merged: https://github.com/kubernetes/utils/pull/267

**Comment by iancoolidge** [comment]
> I will begin looking at it more seriously once this is merged: [kubernetes/utils#267](https://github.com/kubernetes/utils/pull/267)

Hi Kevin, my plan now is actually to try to merge https://github.com/kubernetes/kubernetes/pull/113744 first,
and then to re-do https://github.com/kubernetes/utils/pull/267 with minimal changes, 
and then to do another PR to k/k to re-vendor k/utils and use the library.

So, if you can help by taking a look at https://github.com/kubernetes/kubernetes/pull/113744 first.

**Comment by iancoolidge** [comment]
/remove-lifecycle stale

2nd PR out for review @ https://github.com/kubernetes/kubernetes/pull/115359
